exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/Prel.java

```diff
@@ -56,7 +56,13 @@ public boolean useAbstractConvertersForConversion(RelTraitSet fromTraits,
   SelectionVectorMode getEncoding();
   boolean needsFinalColumnReordering();
 
-  default Prel addImplicitRowIDCol(List<RelNode> children) {
+  /**
+   * If the operator is in Lateral/Unnest pipeline, then it generates a new operator which knows how to process
+   * the rows accordingly during execution.
+   * eg: TopNPrel -> SortPrel and LimitPrel
+   * Other operators like FilterPrel, ProjectPrel etc will add an implicit row id to the output.
+   */
+  default Prel getRelForLateralUnnestPipeline(List<RelNode> children) {
     throw new UnsupportedOperationException("Adding Implicit RowID column is not supported for " +
```

Comment:
This function name is changed from `addImplicitRowIdCol` to `getRelForLateralUnnestPipeline` but from comments it looks like it does both the functionality. Based on my understanding for all operators except `TopNPrel` it's actually adding the `implicitRowIdCol` information, but since topN will not be supported inside subquery and needs to be replaced by PartitionLimit + Sort we are transforming it inside this function. And the new limit and Sort will have the information about the implicitRowIdCol. If this is true, then we should keep the name same as before and in TopN as you have done calling separate private function `transformTopNToSortAndLimit` is fine.

Reply:
changed the name of the function to prepareForLateralUnnestPipeline.